Automattic/wp-calypso
# Use matching version ranges in `composite-checkout`

`composite-checkout` has a dependency on `emotion`, which is composed of multiple packages. However, it (probably inadvertently) depended on multiple versions of `emotion`.

This would not ordinarily be a problem, as `npm` would be able to resolve a single version that matched all requirements. However, this was not possible here, since packages were being required with individual pinned versions, rather than version ranges.

This PR handles the situation in two ways:
- It matches all `emotion` packages to the same logical version.
- It specifies `emotion` packages through a range, rather than an individual pinned version. This is a best practice for packages that are intended to be used elsewhere, as it leaves version resolution to the client, which may be needing `emotion` elsewhere too.

It also removes the `emotion` dependencies from `client/` that are no longer needed there.

#### Changes proposed in this Pull Request

* Switch to version ranges for `emotion` and other dependencies in `composite-checkout` and `composite-checkout-wpcom`.
* Increment versions for `composite-checkout` and `composite-checkout-wpcom`.
* Update `package-lock.json` with the results of the resolution.
* Remove some `emotion` dependencies from `client/`.

#### Testing instructions

Ensure that `composite-checkout` and `composite-checkout-wpcom` continue to work correctly.
